src/twisted/internet/test/test_endpoints.py

```diff
@@ -3259,6 +3259,23 @@ def test_sslNoTrailingNewlinePem(self):
         ctx = server._sslContextFactory.getContext()
         self.assertIsInstance(ctx, ContextType)
 
+    @skipIf(skipSSL, skipSSLReason)
+    def test_sslWithCustomCipher(self):
+        """
+        A cipher list is supported, using the OpenSSL format.
+        The colon (:) from the OpenSSL format is replaced with a comma (,).
+        """
+        reactor = object()
+        server = endpoints.serverFromString(
+            reactor,
+            "ssl:1234:backlog=12:privateKey=%s:"
+            "certKey=%s:cipher=ECDHE-ECDSA-AES128-GCM-SHA256,ECDHE-RSA-AES128-GCM-SHA256,ALL,!ADH"
```

Comment:
I see my previous comment was marked as "resolved" , but I don't see any info about why no change is required.

Do we need `backlog` in this test? Will the test fail if backlog is not provided ?

Is there any value in having both `ECDHE-ECDSA-AES128-GCM-SHA256` and `ECDHE-RSA-AES128-GCM-SHA256` suites in this list ?

I still think that we should look into a minimal list, that covers all the available OpenSSL options ... something like `ALL:!ADH:@STRENGTH:+RSA:-DSA:SHA1+DES`

I can see that this is valid

```
$ openssl ciphers 'ALL:!ADH:@STRENGTH:+RSA:-DSA:SHA1+DES'
```

-----------

I am note sure that this test provides good coverage.

It's important to have

For example, if I remove these lines from the branch, would this test fail ?

```
```suggestion

    if cipher:
        cipherBytes = cipher.replace(",", ":").encode("ascii")
        cf.getContext().set_cipher_list(cipherBytes)
```

--------

I still think that this test is valid.
I also think that the only way to check which ciphers were set, is to do a full handshake ... and this might be too much for this test.

But it's important to add a note about this limitation in the docstring of the test.

---------

What happens if I pass a string like `ssl:1234:cipher=bla,bla-bla`

I think that we need a test to check how the errors are reported.

I guess that the error here would be "SSL routines - no cipher match" ... but is best to have a test in which we describe that we don't have any custom error message and the OpenSSL low-level exception is raised.

I don't know if we want that.

I guess that it would be nice to handle the OpenSSL exception and show a better error message

Reply:
[#fe8f46a](https://github.com/twisted/twisted/pull/12137/commits/fe8f46a3f91f6efc12a73f7c6fab3fb494e17c89), [#53b0d41](https://github.com/twisted/twisted/pull/12137/commits/53b0d41b6a8aa4b90edc2365739031c951e0db13)